lib/index.js

```diff
@@ -215,13 +202,64 @@ function lowerCaseObjectKeys(obj) {
   return newObject;
 }
 
+function validateDate(date, key) {
+  // created & expires are not required
+  if((date === undefined) || (date === null)) {
+    return date;
+  }
+  if(date instanceof Date) {
+    return (Math.round(date.getTime() / 1000));
+  } else if(Number.isInteger(date)) {
+    // we need to accept that the integer is a unix time stamp
+    return date;
+  } else if(typeof(date) === 'string') {
+    // we assume this is a unix time stamp too
+    const number = Number(date);
+    if(!Number.isNaN(number)) {
+      return number;
+    }
+  }
```

Comment:
Some style fixes here -- but also, I think we should require a string to be an integer. I think we've also got a problem here with assuming `Math.round()` vs. Math.floor/Math.ceil. I think we want to use `Math.ceil` for `created` and `Math.floor`for expires if we want to be safe.

```suggestion
  if(date instanceof Date) {
    return (Math.round(date.getTime() / 1000));
  }
  if(Number.isInteger(date)) {
    // we assume the integer is a unix time stamp
    return date;
  }
  if(typeof date === 'string') {
    // we assume the string is a unix time stamp if
    // the whole string represents an integer
    const number = Number(date);
    if(Number.isInteger(number)) {
      return number;
    }
  }
```

Reply:
Hey,

I did the corrections here: https://github.com/digitalbazaar/http-signature-header/pull/17/commits/a0289fdf1b22c3822b4f66c187550596c363853e

one minor issue: `Math.ceil` can result in a timestamp that is 1 second in the future which creates an error so I used `Math.floor` instead.